### ISSUE TITLE
Shader code editor is closed on escape.

### ISSUE THREAD

**Issue by spongeboburu** [godotengine/godot#20738]
**Godot version:**
3.1-dev

**OS/device including version:**
Fedora 28, 64-bit

**Issue description:**
Pressing escape while in shader code editor will close it. This is quite annoying since escape is also used to cancel the code completion popup. 

**Steps to reproduce:**
While editing shader code, press escape.

**Comment by lekoder** [comment]
I identified the code that caused this (2ae2735), but outright removal will be probably considered by regression. Will attempt to disable only for shader editor.

**Comment by lekoder** [comment]
This is a duplicate of #17159 that should be supposedly fixed by #16947 ; possible regression.

**Comment by akien-mga** [comment]
CC @Faless @Zylann - can someone confirm that #16947 did fix #17159 back in March?

**Comment by lekoder** [comment]
I just compiled 201d2d722 (c53126219 did not build) and it is affected by the issue.

**Comment by lekoder** [comment]
Even if you are inside code completion of completely different editor, the ESC key closes all the panels below. Shouldn't code completion cancel consume the input event?

**Comment by Zylann** [comment]
When I suggested that feature a while ago it was under the assumption that *ESC should work with context awareness* (and I believe they should https://github.com/godotengine/godot/issues/9069#issuecomment-398042162). So no, ESC alone should never have been a shortcut to close the bottom panel in any context like it does here, because that's indeed annoying. But the idea was, like QtCreator and Sublime Text, if nothing else catches the input, then it can close it. That idea stemmed first from the fact that everytime I test a scene, it opens the console panel, which is often wanted for debugging. But once the session is finished, I might want to close that panel if I don't need it after that test (so toggling an editor option or reaching for the button with mouse everytime is in turn annoying).

There is an alternative though: I noticed VSCode also has an issue with ESC, and I could not make it cleanly handle it to close its bottom panel (as per issue https://github.com/Microsoft/vscode/issues/48122). A compromise was to have a shortcut Shift+ESC.
